src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
+            {
+                Array dataArray = (Array)(value);
+                ret += "(";
```

Comment:
Would you like to address this comment as well (at least in your newly added method `HandleNamedDataHashValue`)?